pypsa/linopt.py

```diff
@@ -672,9 +670,51 @@ def read_until_break(f):
             constraints_dual, objective)
 
 
+def run_and_read_cplex(n, problem_fn, solution_fn, solver_logfile,
+                        solver_options, warmstart=None, store_basis=True):
+    """
+    Solving function. Reads the linear problem file and passes it to the cplex
+    solver. If the solution is sucessful it returns variable solutions and
+    constraint dual values. Cplex must be installed for using this function
+
+    """
+    import cplex
+    m = cplex.Cplex()
+    out = m.set_log_stream(solver_logfile)
+    if solver_options is not None:
+        for key, value in solver_options.items():
+            getattr(m.parameters, key).set(value)
+    m.read(problem_fn)
+    if warmstart:
+        m.start.read_basis(warmstart)
+    m.solve()
+    is_lp = m.problem_type[m.get_problem_type()] == 'LP'
+
+    termination_condition = m.solution.get_status_string()
+    if 'optimal' in termination_condition:
+        status = 'ok'
+        termination_condition = 'optimal'
+    else:
+        status = 'warning'
+
+    if (status == 'ok') and store_basis and is_lp:
+        n.basis_fn = solution_fn.replace('.sol', '.bas')
+        m.solution.basis.write(n.basis_fn)
+
+    objective = m.solution.get_objective_value()
+    sol = pd.Series(m.solution.get_values(), m.variables.get_names())
```

Comment:
```suggestion
    sol = pd.Series(m.solution.get_values(), m.variables.get_names()).pipe(set_int_index)
```
I was getting string "x101" otherwise. I am not sure whether this was different for you. Then it might be a version issue.

Reply:
Totally right, my test failed me, I had a wrong setting!